lib/easyify.js

```diff
@@ -14,6 +15,8 @@ module.exports = function easyify (keyFn) {
     return function makeKey (id) {
       let y
       if (isFeed(id)) {
+        if (id.startsWith('ssb:feed/ed25519')) id = toFeedSigil(id)
```

Comment:
:fire: is a behaviour change that needs attention.

**Problem** we want to be able to make shared DM keys between an author + a foreign feedId. This `easify` method is used for that, and gives you a method so you can go `makeKey(foreignFeedId)`. It expects an id that it can run `bfe.encode` on, which assumes a lot about the shape of feedIds

Currently it supports:
- :heavy_check_mark: sigil feed ids (classic @)
- :heavy_check_mark: NEW ssb-uri's of form `'ssb:feed/gabbygrove-v1/kQzPN-jCA4CZtthae9UZfQJZY2LjCEPfE-3S7x_iEkk='`
- :x: classic ssb-uri of form `'ssb:feed/ed25519/kQzPN-jCA4CZtthae9UZfQJZY2LjCEPfE-3S7x_iEkk='`
    - unfortunately `bfe.encode` uses `ssb-uri2` to decompose URIs, which is not aligned with `bfe` type_format definitions (for example this feed type = `classic` in bfe, and I also note `ssb-uri2` knows nothing about `bamboo`), and insists on doing some "validation" of it's own 
    - this means that if you put this feed format into `bfe.encode` it, you'll get past `ssb-uri2` but then hit a throw with "unknown format" when you hit `ssb-bfe` trying to use that `ed25519` format
    - if you try to put `'ssb:feed/classic/kQzPN-jCA4CZtthae9UZfQJZY2LjCEPfE-3S7x_iEkk='`` in, that also throws with "unknown format" because `ssb-uri2` doesn't know what type `classic` means

**Solution** : 
- quick, dirty :vomit: (this PR) - if this feed type is detected, shunt it into sigil format, which allows it to side-step all this throwing, as they both seem to be fine with sigils....
- :star_struck: (future) - align `ssb-uri2` and `ssb-bfe` definitions
    - ideally `ssb-uri2` consumes `ssb-bfe-spec/bfe.json` as well and uses that to inform definitions
    - it may want to also offer support for some "additional" aliases like `ssb:feed/ed25519` => `ssb:feed/classic`

Reply:
gonna leave a comment in code